plugin/trino-teradata/src/main/java/io/trino/plugin/teradata/TeradataConfig.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.teradata;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.trino.plugin.jdbc.BaseJdbcConfig;
+
+public class TeradataConfig
+        extends BaseJdbcConfig
+{
+    private String logMech = "TD2";
+    private TeradataCaseSensitivity teradataCaseSensitivity = TeradataCaseSensitivity.CASE_SENSITIVE;
+
+    public String getLogMech()
+    {
+        return logMech;
+    }
+
+    @Config("logon-mechanism")
+    @ConfigDescription("Specifies the logon mechanism for Teradata (default: TD2). Use 'TD2' for TD2 authentication.")
+    public TeradataConfig setLogMech(String logMech)
```

Comment:
You can use enum instead of String here. Then, you can remove "Unsupported logon mechanism" check from TeradataClientModule.